index.js

```diff
@@ -15,23 +16,50 @@ const map = {
 const regex = /<pre><code class="(.*)?">([\s\S]*?)<\/code><\/pre>/igm;
 const captionRegex = /<p><code>(.*?)\s(.*?)\n([\s\S]*)<\/code><\/p>/igm;
 
-const baseDir = hexo.base_dir;
 const rootPath = hexo.config.root || '/';
-const prismDir = path.join(baseDir, 'node_modules', 'prismjs');
-const prismThemeDir = path.join(prismDir, 'themes');
+
+const themeRegex = /^prism-(.*).css$/;
+const prismDir = require.resolve('prismjs/package.json').replace('package.json', '');
```

Comment:
Had thought about using require.resolve("prismjs") here as well, the only problem there is that this could change if prism.js changes their build chain and e.g. set the main in package.json to something like dist/prism.min.js.

The path.parse is definitely much cleaner than replacing the end of the string though! Did not know about this function.

I have seen that it was previously used to resolve the plugin path. Please check the changes i have made and let me know what you think about it, it get's rid of this line completely.